module/mail/mail.py

```diff
@@ -0,0 +1,238 @@
+        check_area = tuple([area[0], area[3] + 3, area[2], area[3] + 5])
+        im = rgb2gray(crop(image, check_area))
+        return True if np.mean(im) < 182 else False
+
+    def _mail_collect_one(self, item, skip_first_screenshot=True):
+        """
+        Executes the collecting sequence on
+        one mail item
+
+        Args:
+            item (Item):
+            skip_first_screenshot (bool):
+        """
+        btn = item._button
+        click_timer = Timer(1.5, count=3)
+        while 1:
+            if skip_first_screenshot:
+                skip_first_screenshot = False
+            else:
+                self.device.screenshot()
+
+            if click_timer.reached():
+                if not self._mail_selected(btn, self.device.image):
+                    self.device.click(btn.move((350, 0)))
+                    click_timer.reset()
+                    continue
+                click_timer.reset()
+            if self.appear_then_click(MAIL_COLLECT, offset=(20, 20), interval=3):
+                click_timer.reset()
+                continue
+            if self.appear(GET_ITEMS_1, offset=(30, 30), interval=1):
+                self.device.click(MAIL_COLLECT)
+                click_timer.reset()
+                continue
+            if self.appear(GET_ITEMS_2, offset=(30, 30), interval=1):
+                self.device.click(MAIL_COLLECT)
+                click_timer.reset()
+                continue
+            if self.handle_info_bar():
+                click_timer.reset()
+                continue
+
+            # End
+            if self.appear(MAIL_COLLECTED, offset=(20, 20)):
+                break
+
+    def _mail_collect(self, skip_first_screenshot=True):
+        """
+        Executes the collecting sequence to
+        applicable mail items
+
+        Args:
+            skip_first_screenshot (bool):
+        """
+        for _ in range(5):
+            if skip_first_screenshot:
+                skip_first_screenshot = False
+            else:
+                self.device.screenshot()
+
+            collectable = self._mail_get_collectable()
+            if not collectable:
+                break
+
+            self._mail_collect_one(collectable[0])
+
+        logger.info('Finished collecting all applicable mail')
+
+    def mail_run(self, collect=True, delete=False):
+        """
+        Collects mail rewards
+
+        Args:
+            collect (bool):
+            delete (bool):
+        """
+        if not collect:
+            return False
+        logger.info('Mail run')
+
+        self._mail_enter(delete)
+        self._mail_collect()
+        self._mail_exit()
+
+    def run(self):
+        """
+        Pages:
+            in: Any page
+            out: page_main, may have info_bar
+        """
+        self.ui_goto(page_main)
+        self.mail_run(collect=self.config.Mail_Collect,
+                      delete=self.config.Mail_Delete)
+        self.config.task_delay(success=True)
```

Comment:
Remove `self.config.task_delay(success=True)`

Reply:
Missed that thanks!